src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -218,9 +218,14 @@ private void CollectAndVisitConcatArguments(BoundBinaryOperator originalOperator
                 // need to consider whether to do so. The visiting logic in the parent function here depends on only one argument being added for a compound assignment
                 // left read, so if we ever do introduce optimizations here that result in more than one argument being added to destinationArguments, we'll need to adjust
                 // that logic.
-                Debug.Assert(visitedCompoundAssignmentLeftRead is
-                    not (BoundCall or BoundConversion { ConversionKind: ConversionKind.Boxing, Type.SpecialType: SpecialType.System_Object, Operand.Type.SpecialType: SpecialType.System_Char })
-                    and { ConstantValueOpt: null });
+#if DEBUG
+                var precedingArgument = destinationArguments.Count > 0 ? destinationArguments[^1] : null;
+                var (singleConcatArgument, nestedConcatArguments) = SimplifyConcatArgument(visitedCompoundAssignmentLeftRead, ref precedingArgument, ref concatMethods);
+                // Simplify should have done no work and just returned the original argument
+                Debug.Assert(ReferenceEquals(singleConcatArgument, visitedCompoundAssignmentLeftRead));
+                Debug.Assert(precedingArgument is null || ReferenceEquals(precedingArgument, destinationArguments[^1]));
```

Comment:
>precedingArgument is null [](http://example.com/codeflow?start=29&length=25)

Is it Ok for this value to become `null` when it was not `null` before the `SimplifyConcatArgument` call?